dspace-api/src/main/java/org/dspace/discovery/indexobject/ClaimedTaskIndexFactoryImpl.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery.indexobject;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.common.SolrInputDocument;
+import org.dspace.core.Context;
+import org.dspace.discovery.indexobject.factory.ClaimedTaskIndexFactory;
+import org.dspace.discovery.indexobject.factory.WorkflowItemIndexFactory;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.xmlworkflow.storedcomponents.ClaimedTask;
+import org.dspace.xmlworkflow.storedcomponents.service.ClaimedTaskService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Factory implementation for indexing/retrieving claimed tasks in the search core
+ * @author Kevin Van de Velde (kevin at atmire dot com)
+ */
+public class ClaimedTaskIndexFactoryImpl extends IndexFactoryImpl<IndexableClaimedTask>
+        implements ClaimedTaskIndexFactory {
+
+    @Autowired
+    protected ClaimedTaskService claimedTaskService;
+    @Autowired
+    WorkflowItemIndexFactory indexableWorkflowItemService;
+
+    @Override
+    public Iterator<IndexableClaimedTask> findAll(Context context) throws SQLException {
+        final Iterator<ClaimedTask> claimedTasks = claimedTaskService.findAll(context).iterator();
+        return new Iterator<IndexableClaimedTask>() {
+            @Override
+            public boolean hasNext() {
+                return claimedTasks.hasNext();
+            }
+
+            @Override
+            public IndexableClaimedTask next() {
+                return new IndexableClaimedTask(claimedTasks.next());
+            }
+        };
+    }
+
+    @Override
+    public String getType() {
+        return IndexableClaimedTask.TYPE;
+    }
+
+    @Override
+    public SolrInputDocument buildDocument(Context context, IndexableClaimedTask indexableObject)
```

Comment:
It would be useful to add some inline comments to explain where the various part of the solr document are built. E.g. for the `super.buildDocument` explain it adds the IDs, types and calls the plugins. Also explain where the item metadata is included, …

Reply:
Some documentation was added.